internal/provider/resource_auth0_connection_client.go

```diff
@@ -87,12 +87,12 @@ func createConnectionClient(ctx context.Context, data *schema.ResourceData, meta
 	api := meta.(*management.Management)
 
 	connectionID := data.Get("connection_id").(string)
+
+	globalMutex.Lock(connectionID)
+	defer globalMutex.Unlock(connectionID)
+
 	connection, err := api.Connection.Read(connectionID)
 	if err != nil {
-		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
```

Comment:
404 checks should not happen on CREATE to prevent `Root resource was present, but now absent` errors.